overlord/devicestate/devicestate.go

```diff
@@ -848,22 +848,28 @@ func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Mo
 			// Already handled
 			continue
 		}
-		newSnap, ts, err := tasksForEssentialSnap(ctx, st,
+		ts, err := tasksForEssentialSnap(ctx, st,
 			modelSnap.SnapType, current, new, localSnaps, paths,
 			snapRevisions[modelSnap.SnapName()], vSetKeys,
-			deviceCtx, fromChange)
+			tracker, deviceCtx, fromChange)
 		if err != nil {
 			return nil, err
 		}
 		if ts != nil {
 			tss = append(tss, ts)
-			if modelSnap.SnapType == "gadget" {
-				if err := updateNeededSnapsFromTs(ts); err != nil {
-					return nil, err
-				}
-			}
 		}
-		snapsAccountedFor[newSnap] = true
+	}
+
+	// if base is not set, then core will not be returned in the list of snaps
```

Comment:
But then it is a UC16 system, right? And we do not plan to support remodeling from UC16 to anything, so not sure if this bit is needed.

Reply:
Do we prevent remodeling a UC16 device right now? I don't see that as one of the guard clauses in `devicestate.Remodel`

Reply:
We prevent remodels from UC16 to 18+, but I was not sure about remodels changing other snaps. However I found that we have a spread test for that case (tests/core/remodel), so this should be fine. Maybe it is worth adding a sentence to the comment saying that moving to more modern UC is not supported in this case.